src/renderer/vt/paint.cpp

```diff
@@ -461,10 +461,24 @@ using namespace Microsoft::Console::Types;
     //      we'll determine that we need to emit a \b to put the cursor in the
     //      right position. This is wrong, and will cause us to move the cursor
     //      back one character more than we wanted.
-    if (_lastText.X < _lastViewport.RightInclusive())
+    //
+    // GH#1245: This needs to be RightExclusive, _not_ inclusive. Otherwise, we
+    // won't update our internal cursor position tracker correctly at the last
+    // character of the row.
+    if (_lastText.X < _lastViewport.RightExclusive())
     {
         _lastText.X += static_cast<short>(columnsActual);
     }
+    // GH#1245: If we wrote the exactly last char of the row, then we're
+    // probably in the "delayed EOL wrap" state. Different terminals (conhost,

```

Comment:
I don't like the "probably" here. How could we be sure or how does it vary?

Reply:
Yea this shouldn't say "probably", we _are_ in that state.